opentelemetry-sdk/src/metrics/periodic_reader_with_own_thread.rs

```diff
@@ -0,0 +1,798 @@
+            MetricResult, SdkMeterProvider, Temporality,
+        },
+        testing::metrics::InMemoryMetricExporter,
+        Resource,
+    };
+    use async_trait::async_trait;
+    use opentelemetry::metrics::MeterProvider;
+    use std::{
+        sync::{
+            atomic::{AtomicUsize, Ordering},
+            mpsc, Arc,
+        },
+        time::Duration,
+    };
+
+    // use below command to run all tests
+    // cargo test metrics::periodic_reader_with_own_thread::tests --features=testing,experimental_metrics_periodic_reader_no_runtime -- --nocapture
+
+    #[derive(Debug, Clone)]
+    struct MetricExporterThatFailsOnlyOnFirst {
+        count: Arc<AtomicUsize>,
+    }
+
+    impl Default for MetricExporterThatFailsOnlyOnFirst {
+        fn default() -> Self {
+            MetricExporterThatFailsOnlyOnFirst {
+                count: Arc::new(AtomicUsize::new(0)),
+            }
+        }
+    }
+
+    impl MetricExporterThatFailsOnlyOnFirst {
+        fn get_count(&self) -> usize {
+            self.count.load(Ordering::Relaxed)
+        }
+    }
+
+    #[async_trait]
+    impl PushMetricExporter for MetricExporterThatFailsOnlyOnFirst {
+        async fn export(&self, _metrics: &mut ResourceMetrics) -> MetricResult<()> {
+            if self.count.fetch_add(1, Ordering::Relaxed) == 0 {
+                Err(MetricError::Other("export failed".into()))
+            } else {
+                Ok(())
+            }
+        }
+
+        async fn force_flush(&self) -> MetricResult<()> {
+            Ok(())
+        }
+
+        fn shutdown(&self) -> MetricResult<()> {
+            Ok(())
+        }
+
+        fn temporality(&self) -> Temporality {
+            Temporality::Cumulative
+        }
+    }
+
+    #[test]
+    fn collection_triggered_by_interval_multiple() {
+        // Arrange
+        let interval = std::time::Duration::from_millis(1);
+        let exporter = InMemoryMetricExporter::default();
+        let reader = PeriodicReaderWithOwnThread::builder(exporter.clone())
+            .with_interval(interval)
+            .build();
+        let i = Arc::new(AtomicUsize::new(0));
+        let i_clone = i.clone();
+
+        // Act
+        let meter_provider = SdkMeterProvider::builder().with_reader(reader).build();
+        let meter = meter_provider.meter("test");
+        let _counter = meter
+            .u64_observable_counter("testcounter")
+            .with_callback(move |_| {
+                i_clone.fetch_add(1, Ordering::Relaxed);
+            })
+            .build();
+
+        // Sleep for a duration 5X (plus liberal buffer to account for potential
+        // CI slowness) the interval to ensure multiple collection.
+        // Not a fan of such tests, but this seems to be the only way to test
+        // if periodic reader is doing its job.
+        // TODO: Decide if this should be ignored in CI
+        std::thread::sleep(interval * 5 * 20);
```

Comment:
yes sleep will slow down the CI. Instead of sleep, we can have a  loop that periodically checks this condition as below (not syntactically correct):
```
while i.load(Ordering::Relaxed) < 5 && elapsed_time < 5*20 ms) 
{
}
```

Or better to skip this test for the CI. Not a blocker for this PR.